### ISSUE TITLE
[Feature Request] Deepl translation support

### ISSUE THREAD

**Issue by IkariDevGIT** [SillyTavern/SillyTavern#568]
**Have you searched for similar [requests](https://github.com/SillyTavern/SillyTavern/issues?q=)?**
Yes

**Is your feature request related to a problem? Please describe.**
I like the translation extension, but google is like.. bad

**Describe the solution you'd like**
Implement deepl api into the translation extension

**Describe alternatives you've considered**
i couldnt think of any

**Comment by Woisek** [comment]
As much I would also like to have deepl as translator, I don't think it would be free to use ... 😐

**Comment by IkariDevGIT** [comment]
@Woisek scroll down, there you have it... they have a free api https://www.deepl.com/pro-api

**Comment by Woisek** [comment]
> @Woisek scroll down, there you have it... they have a free api https://www.deepl.com/pro-api

Yes, they have that. But first, in that case every user has to register for that and second, do you think, that 500000 characters per month is really sufficient for something like a text generator ... ?
If one does occasionally use deepl for small texts or so, yes, it will be sufficient. But every hc user using it for role play will probably reach the 500000 chars after a day ... at best.
Just saying ...

**Comment by IkariDevGIT** [comment]
@Woisek yea, but its still worth implementing, many users use for example the OpenAI API, and it also costs money.. so if someone wants a REALLY good translation they WILL pay for and use it(probably)(this is about 500.000 words btw https://pastebin.com/raw/PvGm96qk)

**Comment by Woisek** [comment]
Yeah, I'm all for that, I have no problem, I was just saying.
I'm using deepl myself sometimes when chatting, because English is not my native language. Sometimes I also would like to chat completely in German, but I'm OK with this situation, too.
But generally you are completely right, having this option would be an good addition, whether one uses it or not. 🙂

**Comment by IkariDevGIT** [comment]
@Woisek sorry if it came out agressive, my main language is also not english, its german 😆

**Comment by Woisek** [comment]
> @Woisek sorry if it came out agressive, my main language is also not english, its german 😆

Na sehr fein, Dann können wir github endlich übernehmen ... 😁
No problem, I didn't took it aggressively. Like I said ... just saying. 🙂

**Comment by Woisek** [comment]
@Cohee1207 Wonderful 👍 ... any infos on when it will be merged into main? 🙂
